gcp_airflow_foundations/base_class/template_ingestion_options_config.py

```diff
@@ -0,0 +1,23 @@
+from dacite import Config
+from pydantic import validator, root_validator
+from pydantic.dataclasses import dataclass
+
+from typing import List, Optional
+import regex as re
+
+from gcp_airflow_foundations.enums.schema_source_type import SchemaSourceType
+
+
+@dataclass
+class TemplateIngestionOptionsConfig:
+    """
+    Attributes:
+        ingest_all_tables: if true, ingest all tables from source
+        ingestion_name: name to use for DAG id, should correspond to the relevant partition of the source cut out by the regex expression provided
+        dag_creation_mode: if "SOURCE", then one DAG per source is created. if "TABLE", one DAG per table.
```

Comment:
Would it be easier to create an enum DagCreationMode, this will make validations simpler (just out of the box)

Reply:
Also, for TableConfig, this is on the root level right? So it would make sense to move it to the Root level for this too

Reply:
for each template, we should be able to provide a separate regular expression + ingestion name, no?

Reply:
that way, can provide a bunch of sections of the data within 1 config file